packages/block-library/src/image/index.php

```diff
@@ -28,12 +28,23 @@ function render_block_core_image( $attributes, $content, $block ) {
 		return '';
 	}
 
-	if ( isset( $attributes['data-id'] ) ) {
+	if ( isset( $attributes['id'] ) ) {
 		// Adds the data-id="$id" attribute to the img element to provide backwards
 		// compatibility for the Gallery Block, which now wraps Image Blocks within
 		// innerBlocks. The data-id attribute is added in a core/gallery
 		// `render_block_data` hook.
-		$p->set_attribute( 'data-id', $attributes['data-id'] );
+		$id = $attributes['id'];
+		$p->set_attribute( 'data-id', $id );
```

Comment:
The side effect of this is that the image block will now always have `data-id`, whereas in trunk it's only added when the image block is in a gallery. Perhaps there's another way to detect when the image block is nested that doesn't rely on the extra block attribute.

May also need to add some back compat for the possibility that third parties filter the `data-id` block attribute. 🤔